packages/dom/src/dom.js

```diff
@@ -209,6 +218,8 @@ export function getRectangleFromRange( range ) {
 	// See: https://stackoverflow.com/a/6847328/995445
 	if ( ! rect ) {
 		const padNode = document.createTextNode( '\u200b' );
+		// Do not modify to live range.
+		range = range.cloneRange();
```

Comment:
It reminds me of some unrelated testing I'd been doing with Chrome line-ending bugginess, where `Range#cloneRange` wasn't actually avoiding mutations to the original:

https://codepen.io/aduth/pen/WPjQoW

(See comment: "// WIP: Buggy, collapses spaces (mutates original).")

I never did figure out what I was seeing with my demos. I raise only in case there's still a worry about mutating the cloned value.

Reply:
I've not tested the Codepen, but when you make this change in master, this test will also fail:

https://github.com/WordPress/gutenberg/blob/917338c56c4414642ab744a112ec3c8490316493/packages/e2e-tests/specs/adding-blocks.test.js#L64-L71

It is true that cloning the range in `isEdge` will also prevent this issue, but I think it is safe to clone here is well in case it is called with a live range.

In my testing `insertNode` doesn't modify the original range.